### PR TITLE
Add some docs, but the SetCookie and Cookie are a bit not clear

### PR DESCRIPTION
**Pull request by harikt**
Hi @simensen , 

I started to create some docs for the Readme. Somethings that came into my mind are 

1 ) There is a Cookies class 
2 ) Which contains collection of Cookie objects.
3 ) Now Cookies class have a method to convert them and attach to PSR 7 Requests.
4 ) There is a SetCookie class which have some what similarities with that of setcookie method of php. 

May be the purpose of Cookies is just to read and `SetCookies` is what we need to make use of when we need to send to response ?

Thank you

### CODE REVIEW

**Comment by simensen** [comment]
Yes, Cookies is on the request side (reading Cookies from the Request) and SetCookie is for the Response side. I hadn't finished fleshing out the SetCookie side yet so it probably isn't easy to follow. :)

Thanks for your help!

File: README.md [hunk 1]
```diff
@@ -2,3 +2,71 @@ FIG Cookies
...
+
+This will return a `Dflydev\FigCookies\Cookie` object via which you can get
+name, value etc via `getName()` and `getValue()` respectively.
+
+## Change cookie value
+
+You can change the value of a cookie via `withValue` method.
+
+```php
+$cookie->withValue('some value');
```
Comment (simensen) [inline README.md]: You'd have to reassign using the immutable stuff.

```php
$cookie = $cookie->withValue('some value');
```

Comment (simensen) [inline README.md]: Also, it should be noted that simply changing a cookie value doesn't update values in cookies, either. The two ways to "change (or set) a cookie" would probably be two options like this:

```php
// Get the "token" cookie and set a new value.
$cookie = $cookies->get('token')->withValue('some new value');

// Replace the "token" cookie in the collection of Cookies.
$cookies = $cookies->with($cookie);
```

... or...

```php
// Create a "token" cookie and set a new value.
$cookie = Cookie::create('token')->withValue('some value');

// Set the "token" cookie in the collection of Cookies.
$cookies = $cookies->with($cookie);
```

File: README.md [hunk 2]
```diff
@@ -2,3 +2,71 @@ FIG Cookies
...
+use Dflydev\FigCookies\Cookie;
+
+$cookie = new Cookie('name');
+// or
+// $cookie = Cookie::create('name');
+
+$cookie->withValue('some value');
+```
+
+## Setting cookie to header
```
Comment (simensen) [inline README.md]: This section is wrong; I'm working on finishing the API for this. Right now it is going to be:

```php

// $response = a psr7response

// Create a SetCookies collection from a response.
$setCookies = SetCookies::fromResponse($response);

// Create a SetCookie.
SetCookie::create('lu')
    ->withValue('Rg3vHJZnehYLjVg7qi3bZjzg')
    ->withExpires('Tue, 15-Jan-2013 21:47:38 GMT')
    ->withMaxAge(500)
    ->withPath('/')
    ->withDomain('.example.com')
    ->withSecure(true)
    ->withHttpOnly(true)
;

// Set the SetCookie in the SetCookies collection.
$setCookies = $setCookies->with($setCookie);

// Render the SetCookies collection into the response.
$response = $setCookies->renderIntoSetCookieHeader($response);
```

**Comment by simensen** [comment]
Hey @harikt, I'm going to hold off merging this and you might hold off making more changes to it. I totally appreciate it, but the final API might be a lot nicer. I'm currently working on some facades that will make working with this lower-level value objects easier.

For example:

```php
$cookie = Cookie::create('lu', 'Rg3vHJZnehYLjVg7qi3bZjzg');
$request = FigCookies::setRequestCookie($request, $cookie);
```

... this will be shorthand for...

```php
$cookie = Cookie::create('lu', 'Rg3vHJZnehYLjVg7qi3bZjzg');
Cookies::fromRequest($request)
    ->with($cookie)
    ->renderIntoCookieHeader($request)
;
```


... and...

```php
$request = FigCookies::setRequestCookieFromStrings(
    $request,
    'lu',
    'Rg3vHJZnehYLjVg7qi3bZjzg'
);
```

... to do without using any of the primitives all together.

**Comment by harikt** [comment]
sure @simensen .

Was away from keyboard. Internet outage and some other stuffs.

Thank you.

**Comment by simensen** [comment]
Hi @harikt, let me know what you think of the new README docs. I'm going to close this for now.

**Comment by harikt** [comment]
First quick look, looks good. Will come up with suggestions later.

Thank you.

**Pull request closed without merging**